From 10e987cc33e6e2c2009de27e7dd62ac261de3e6c Mon Sep 17 00:00:00 2001
From: Weijie Gao <weijie.gao@mediatek.com>
Date: Wed, 10 Dec 2025 14:51:11 +0800
Subject: [PATCH 17/23] board: mediatek: fix coding style of mtd fdt detection

Fix coding style of mtd fdt detection

Change-Id: Ifb234100f9163c9298ec387a05b226bf3ac135e9
Signed-off-by: Weijie Gao <weijie.gao@mediatek.com>
CR-Id: WCNCR00180092
---
 board/mediatek/common/bootmenu_mtd.c |  3 +--
 board/mediatek/common/bootmenu_ubi.c |  3 +--
 board/mediatek/common/mtd_helper.c   | 17 ++++++-----------
 3 files changed, 8 insertions(+), 15 deletions(-)

diff --git a/board/mediatek/common/bootmenu_mtd.c b/board/mediatek/common/bootmenu_mtd.c
index c2407925a0e..8a1a3f6b5cb 100644
--- a/board/mediatek/common/bootmenu_mtd.c
+++ b/board/mediatek/common/bootmenu_mtd.c
@@ -179,9 +179,8 @@ void default_boot_set_defaults(void *fdt)
 	int ret;
 
 	ret = mtd_verify_linux_fdt(fdt);
-	if (ret) {
+	if (ret)
 		hang();
-	}
 
 	mtd_boot_set_defaults(fdt);
 }
diff --git a/board/mediatek/common/bootmenu_ubi.c b/board/mediatek/common/bootmenu_ubi.c
index 7aa8f88d137..dd2e403cc07 100644
--- a/board/mediatek/common/bootmenu_ubi.c
+++ b/board/mediatek/common/bootmenu_ubi.c
@@ -176,9 +176,8 @@ void default_boot_set_defaults(void *fdt)
 	int ret;
 
 	ret = mtd_verify_linux_fdt(fdt);
-	if (ret) {
+	if (ret)
 		hang();
-	}
 
 	mtd_boot_set_defaults(fdt);
 }
diff --git a/board/mediatek/common/mtd_helper.c b/board/mediatek/common/mtd_helper.c
index 3fc32e47e3c..f9c691a14b7 100644
--- a/board/mediatek/common/mtd_helper.c
+++ b/board/mediatek/common/mtd_helper.c
@@ -1738,13 +1738,11 @@ void mtd_boot_set_defaults(void *fdt)
 
 int mtd_verify_linux_fdt(void *fdt)
 {
-	struct mtd_info *mtd_nmbm = NULL;
-	bool uboot_is_nmbm = false;
-	bool fdt_has_nmbm = false;
-	int ubi_partition_count = 0;
+	int ubi_partition_count = 0, partitions_node = -1;
+	bool uboot_is_nmbm = false, fdt_has_nmbm = false;
 	int nodeoffset, depth, len;
+	struct mtd_info *mtd_nmbm;
 	const void *prop;
-	int partitions_node = -1;
 
 	if (!fdt)
 		return 0;
@@ -1760,14 +1758,12 @@ int mtd_verify_linux_fdt(void *fdt)
 	depth = 0;
 	while ((nodeoffset = fdt_next_node(fdt, nodeoffset, &depth)) >= 0) {
 		prop = fdt_getprop(fdt, nodeoffset, "mediatek,nmbm", &len);
-		if (prop) {
+		if (prop)
 			fdt_has_nmbm = true;
-		}
 
 		prop = fdt_getprop(fdt, nodeoffset, "compatible", &len);
-		if (prop && strstr((const char *)prop, "fixed-partitions")) {
+		if (prop && strstr((const char *)prop, "fixed-partitions"))
 			partitions_node = nodeoffset;
-		}
 	}
 
 	/* Count UBI partitions (partitions with compatible = "linux,ubi") */
@@ -1775,9 +1771,8 @@ int mtd_verify_linux_fdt(void *fdt)
 		int child;
 		fdt_for_each_subnode(child, fdt, partitions_node) {
 			prop = fdt_getprop(fdt, child, "compatible", &len);
-			if (prop && strstr((const char *)prop, "linux,ubi")) {
+			if (prop && strstr((const char *)prop, "linux,ubi"))
 				ubi_partition_count++;
-			}
 		}
 	}
 
-- 
2.45.2

